src/sources/docker_logs/mod.rs

```diff
@@ -224,8 +234,86 @@ impl SourceConfig for DockerLogsConfig {
+    fn outputs(&self, global_log_namespace: LogNamespace) -> Vec<Output> {
+        let log_namespace = global_log_namespace.merge(self.log_namespace);
+        let schema_definition = BytesDeserializerConfig
+            .schema_definition(log_namespace)
+            .with_source_metadata(
+                Self::NAME,
+                Some(LegacyKey::Overwrite(owned_value_path!(self
+                    .host_key
+                    .as_str()))),
+                &owned_value_path!("host"),
+                Kind::bytes().or_undefined(),
+                Some("host"),
+            )
+            .with_source_metadata(
+                Self::NAME,
+                Some(LegacyKey::Overwrite(owned_value_path!(CONTAINER))),
+                &owned_value_path!(CONTAINER),
+                Kind::bytes(),
+                None,
+            )
+            .with_source_metadata(
+                Self::NAME,
+                Some(LegacyKey::Overwrite(owned_value_path!(IMAGE))),
+                &owned_value_path!(IMAGE),
+                Kind::bytes(),
+                None,
+            )
+            .with_source_metadata(
+                Self::NAME,
+                Some(LegacyKey::Overwrite(owned_value_path!(NAME))),
+                &owned_value_path!(NAME),
+                Kind::bytes(),
+                None,
+            )
+            .with_source_metadata(
+                Self::NAME,
+                Some(LegacyKey::Overwrite(owned_value_path!(CREATED_AT))),
+                &owned_value_path!(CREATED_AT),
+                Kind::timestamp(),
+                None,
+            )
+            .with_source_metadata(
+                Self::NAME,
+                Some(LegacyKey::Overwrite(owned_value_path!("label"))),
+                &owned_value_path!("labels"),
+                Kind::object(Collection::empty().with_unknown(Kind::bytes())).or_undefined(),
+                None,
+            )
+            .with_source_metadata(
+                Self::NAME,
+                Some(LegacyKey::Overwrite(owned_value_path!(STREAM))),
+                &owned_value_path!(STREAM),
+                Kind::bytes(),
+                None,
+            )
+            .with_source_metadata(
+                Self::NAME,
+                parse_value_path(log_schema().timestamp_key())
+                    .ok()
+                    .map(LegacyKey::Overwrite),
+                &owned_value_path!("timestamp"),
+                Kind::timestamp(),
+                Some("timestamp"),
+            )
+            .with_vector_metadata(
+                parse_value_path(log_schema().source_type_key())
+                    .ok()
+                    .as_ref(),
+                &owned_value_path!("source_type"),
+                Kind::bytes(),
+                None,
+            )
+            .with_vector_metadata(
+                None,
+                &owned_value_path!("ingest_timestamp"),
+                Kind::timestamp(),
+                None,
+            );
```

Comment:
Just double checking- Is the reason not using the `.with_standard_vector_source_metadata()` , because the legacy path for timestamp is not `log_schema().timestamp_key()` ?

Reply:
The legacy path _is_ `timestamp_key()` but bizarrely it appears as though it's possible for this source to produce events without a timestamp - we don't replace it with a `now()` or anything just try and parse it from the incoming log.